openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -700,13 +694,13 @@ def check(self, instance):
 
         try:
             # Authenticate and add the instance api to apis cache
-            self.init_api(instance, custom_tags)
+            keystone_server_url = self._get_keystone_server_url(instance)
```

Comment:
In case `keystone_server_url` is `None`, it seems the code will fail here, down the road:
https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/utils/proxy.py#L16

Maybe a check here could help debugging when that happen:

```python
            keystone_server_url = self._get_keystone_server_url(instance)
            if not keystone_server_url:
                self.log.error(...)
                raise IncompleteConfig() # or something in this line
```

Reply:
added inside self._get_keystone_server_url

Reply:
If `keystone_server_url = self._get_keystone_server_url(instance)`, it's not great to let the program continue, and let it fail on the url parser. Would be better to stop asap. What do you think ?

Reply:
? It does not return `None` anymore. If not url is specified and no file is specified then it raises a config error. If a file is specified and the file does not specify the url the parser assumes a default url based on the cloud name and I assume that is part of the config file spec. I added tests for these cases in https://github.com/DataDog/integrations-core/pull/3920/files#diff-2157a9c8363e4f884b333e2cca3bca3cR353

Reply:
Oh cool, my bad, didn't saw the new `IncompleteConfig`.